app/src/main/java/com/keylesspalace/tusky/viewdata/NotificationViewData.kt

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+package com.keylesspalace.tusky.viewdata
+
+import com.keylesspalace.tusky.entity.Notification
+import com.keylesspalace.tusky.entity.Report
+import com.keylesspalace.tusky.entity.TimelineAccount
+
+sealed class NotificationViewData {
+
+    abstract val id: String
+
+    data class Concrete(
+        override val id: String,
+        val type: Notification.Type,
+        val account: TimelineAccount,
+        val statusViewData: StatusViewData.Concrete?,
+        val report: Report?
+    ) : NotificationViewData()
+
+    data class Placeholder(
+        override val id: String,
+        val isLoading: Boolean
+    ) : NotificationViewData()
+
+    fun asStatusOrNull() = (this as? Concrete)?.statusViewData
```

Comment:
I don't really like the fact that an implementation located inside a parent class or interface depends on one or more child classes.

I would either provide no implementation for these 2 methods in the parent interface and implement them properly in each child class (directly returning `null` or the correct field), **or** move these 2 methods out of the interface and turn them into extension methods on `NotificationViewData`.